amplify/backend/function/fetchMotionState/src/utils.js

```diff
@@ -314,11 +314,48 @@ const updateMotionMessagesInDB = async (motionData, motionMessages, flag) => {
   });
 };
 
+const updateMotionFinalizedMessages = async (
+  motionData,
+  motionStateHistory,
+  isDecision = false,
+) => {
+  const didPass = didMotionPass(motionData);
+
+  // Check if the motion passed and the messages have not already been stored in the db
+  if (didPass && !motionStateHistory.hasPassed) {
+    const newMessages = [];
+
+    // only display voting results if a vote has occurred
+    if (motionStateHistory.hasVoted) {
+      newMessages.push('MotionRevealResultMotionWon');
+    }
+
+    if (!isDecision) {
+      newMessages.push('MotionHasPassed');
+    }
+
+    await updateMotionMessagesInDB(motionData, newMessages, 'hasPassed');
```

Comment:
Ideally the `hasPassed` / `hasFailed` / `MotionRevealResultObjectionWon` etc flags would be enum'd or at least Typescripted in some way.

Alas we don't have that (yet) in lambdas